core/FrontController.php

```diff
@@ -615,4 +599,62 @@ private function throwIfPiwikVersionIsOlderThanDBSchema()
             throw new DatabaseSchemaIsNewerThanCodebaseException(implode(" ", $messages));
         }
     }
+
+    private function makeSessionAuthenticator()
+    {
+        $module = Common::getRequestVar('module', self::DEFAULT_MODULE, 'string');
+        $action = Common::getRequestVar('action', false);
+
+        // the session must be started before using the session authenticator,
+        // so we do it here, if this is not an API request.
+        if (SettingsPiwik::isPiwikInstalled()
+            && ($module !== 'API' || ($action && $action !== 'index'))
```

Comment:
Bit unrelated but maybe could also directly exclude to start a session for the opt out iframe see https://github.com/matomo-org/matomo/issues/12540 ?

Reply:
If I remember correctly, the opt-out form uses a nonce & the nonce is stored in a session, so I think it's required for the moment (at least changing that would require a bit more work).

Reply:
👍